Fix FTBFS with -D_FORTIFY_SOURCE=2 compile flag.
By: أحمد المحمودي (Ahmed El-Mahmoudy) <aelmahmoudy@users.sourceforge.net>
Index: slmodem-2.9.11-20080817/modem/modem_datafile.c
===================================================================
--- slmodem-2.9.11-20080817.orig/modem/modem_datafile.c	2008-09-21 10:47:22.000000000 +0200
+++ slmodem-2.9.11-20080817/modem/modem_datafile.c	2008-09-21 11:05:13.000000000 +0200
@@ -48,6 +48,8 @@
 #include <string.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 
 #include <modem_defs.h>
 #include <modem_debug.h>
@@ -111,7 +113,7 @@
 	int ret = -1;
 	int fd;
 
-	fd = open(file_name,O_CREAT|O_WRONLY);
+	fd = open(file_name,O_CREAT|O_WRONLY,(S_IRUSR|S_IWUSR));
 	if(fd < 0)
 		return -errno;
 
